test/integration/main_test.go

```diff
@@ -167,6 +173,9 @@ func TestMappingEventHandling(t *testing.T) {
 			}
 			t.Logf("using trace ID %s", traceID.String())
 
+			wantFilePath := fmt.Sprintf("test-dir/traceID=%s.yaml", traceID)
```

Comment:
```suggestion
			filePath := fmt.Sprintf("test-dir/traceID-%s.yaml", traceID)
```

Also please have consistent names across the tests. You used `filePath` in the other test